src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java

```diff
@@ -165,24 +199,21 @@ public Content getInlineTagOutput(Element holder, DocTree tag, TagletWriter writ
                 if (fileObject == null && fileManager.hasLocation(Location.SNIPPET_PATH)) {
                     fileObject = fileManager.getFileForInput(Location.SNIPPET_PATH, "", v);
                 }
-            } catch (IOException | IllegalArgumentException e) {
+            } catch (IOException | IllegalArgumentException e) { // TODO: test this when JDK-8276892 is integrated
```

Comment:
You are much more knowledgeable in FileManager than I am. My thinking was that while it's probably not an issue with the default file manager, I couldn't guarantee that IllegalArgumentException is not thrown if this code runs with a custom file manager.

Reply:
I was just reading the documented conditions for this `JavaFileManager` method to throw `IllegalArgumentException`. That being said, one of the problems with ILA, and potentially other unchecked exception, is that they can occur from deep within the method that is called. 

FYI, javac has a way of wrapping user-provided components so that it can better handle unexpected exceptions from user-provided code as compared to unexpected exceptions in javac itself. If you're interested, see`javac` `ClientCodeWrapper`.   And, to be clear, I'm *not* suggesting that for javadoc at this time!

Reply:
> That being said, one of the problems with ILA, and potentially other unchecked exception, is that they can occur from deep within the method that is called.

Care to elaborate?